### ISSUE TITLE
Spectra downloading doesn't work

### ISSUE THREAD

**Issue by bdrum** [regata-jinr/RegataDBC#9]
In case of downloading first three containers from №226 only 13 file of 59 will be download for LLI1.

**Comment by bdrum** [comment]
Downloading spectra should be async and faster

**Comment by bdrum** [comment]
Refactor the code with separate task usage for search and download process.
List of files -> List of Search Tasks -> Each task continue with downloading task

**Comment by bdrum** [comment]
Getting spectra should be part of set content form.
It will be based on [nextcloud webdav api](https://docs.nextcloud.com/server/15/developer_manual/client_apis/WebDAV/index.html)

**Comment by bdrum** [comment]
Search request example:

~~~powershell
 curl -X SEARCH 'https://disk.jinr.ru/remote.php/dav'  -u bdrum -H "content-Type: text/xml" --data '<?xml version=\"1.0\" encoding=\"UTF-8\"?>
 <d:searchrequest xmlns:d=\"DAV:\"
     xmlns:oc=\"http://owncloud.org/ns\">
     <d:basicsearch>
         <d:select>
             <d:prop>
                 <d:displayname/>
             </d:prop>
         </d:select>
         <d:from>
             <d:scope>
                 <d:href>files/bdrum/RegataStorage/Spectra</d:href>
                 <d:depth>infinity</d:depth>
             </d:scope>
         </d:from>
         <d:where>
             <d:eq>
                 <d:prop>
                     <d:displayname/>
                 </d:prop>
                 <d:literal>7203231.CNF</d:literal>
             </d:eq>
         </d:where>
    </d:basicsearch>
 </d:searchrequest>'
Enter host password for user 'bdrum':
~~~

Output
~~~xml
<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" 
  xmlns:s="http://sabredav.org/ns" 
  xmlns:cal="urn:ietf:params:xml:ns:caldav" 
  xmlns:cs="http://calendarserver.org/ns/" 
  xmlns:card="urn:ietf:params:xml:ns:carddav" 
  xmlns:oc="http://owncloud.org/ns" 
  xmlns:nc="http://nextcloud.org/ns">
  <d:response>
    <d:href>/remote.php/dav/files/bdrum/RegataStorage/Spectra/2015/12/dji-2/7203231.CNF</d:href>
    <d:propstat>
      <d:prop>
        <d:displayname/>
      </d:prop>
      <d:status>HTTP/1.1 404 Not Found</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>
~~~

**Comment by bdrum** [comment]
## Get list of folder request:

~~~powershell
curl -u bdrum -X PROPFIND https://disk.jinr.ru/remote.php/dav/files/bdrum/RegataStorage
~~~

output:

~~~xml
?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" 
  xmlns:s="http://sabredav.org/ns" 
  xmlns:cal="urn:ietf:params:xml:ns:caldav" 
  xmlns:cs="http://calendarserver.org/ns/" 
  xmlns:card="urn:ietf:params:xml:ns:carddav" 
  xmlns:oc="http://owncloud.org/ns" 
  xmlns:nc="http://nextcloud.org/ns">
  <d:response>
    <d:href>/remote.php/dav/files/bdrum/RegataStorage/</d:href>
    <d:propstat>
      <d:prop>
        <d:getlastmodified>Tue, 26 May 2020 12:42:08 GMT</d:getlastmodified>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <d:quota-used-bytes>0</d:quota-used-bytes>
        <d:quota-available-bytes>53144075232</d:quota-available-bytes
        <d:getetag>&quot;5ecd0ea048f70&quot;</d:getetag>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/remote.php/dav/files/bdrum/RegataStorage/2020/</d:href>
    <d:propstat>
      <d:prop>
        <d:getlastmodified>Tue, 26 May 2020 12:42:08 GMT</d:getlastmodified>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <d:quota-used-bytes>0</d:quota-used-bytes>
        <d:quota-available-bytes>53144075232</d:quota-available-bytes>
        <d:getetag>&quot;5ecd0ea047d46&quot;</d:getetag>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
    <d:propstat>
      <d:prop>
        <d:getcontentlength/>
        <d:getcontenttype/>
      </d:prop>
      <d:status>HTTP/1.1 404 Not Found</d:status>
    </d:propstat>
  </d:response>
<
~~~

## Download file:

~~~powershell
curl -u bdrum -X GET 'https://disk.jinr.ru/remote.php/dav/files/bdrum/Nextcloud%20Manual.pdf' --output Nextcloud.pdf
~~~

## Create folder:

~~~powershell
curl -u bdrum  -X MKCOL 'https://disk.jinr.ru/remote.php/dav/files/bdrum/RegataStorage/2020'
~~~


## Upload file

~~~powershell
curl  -X PUT https://disk.jinr.ru/remote.php/dav/files/bdrum/RegataStorage/2020/Nextcloud.pdf -u bdrum -T .\Nextcloud.pdf
~~~

**Comment by bdrum** [comment]
[Fixed in the latest release](https://github.com/regata-jinr/RegataDBC/commit/02c38cda646617d217e11cdb49411defabc41d3b)
